packages/tuono-router/src/components/RouterContext.tsx

```diff
@@ -1,13 +1,29 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import type { ReactNode } from 'react'
 
 import type { Router } from '../router'
+import type { ServerRouterInfo } from '../types'
+
+export interface ParsedLocation {
+  href: string
+  pathname: string
+  search: Record<string, string>
+  searchStr: string
+  hash: string
+}
+
+export interface RouterContextType {
+  router: Router
+  location: ParsedLocation
+  updateLocation: (loc: ParsedLocation) => void
+}
 
 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-const routerContext = React.createContext<Router>(null!)
+const routerContext = React.createContext<RouterContextType>(null!)
```

Comment:
[Context should have a PascalCase name](https://react.dev/reference/react/createContext)

```suggestion
const RouterContext = React.createContext<RouterContextType>(null!)
```